### PR TITLE
Fix memory highwater and usage reporting

### PR DESCRIPTION
**Pull request by amametjanov**
Fix memory highwater and usage reporting

Fixes E3SM-Project/E3SM#3292

[BFB]

### CODE REVIEW

**Comment by amametjanov** [comment]
On Anvil, with `SMS.ne4_ne4.FC5AV1C-L.anvil_intel.20191211`, memory highwater and usage numbers across 3 jobs are now stable: variance is under 1%
```
 memory_write: model date =   00010102       0 memory =    2422.43 MB (highwater)       1453.65 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010103       0 memory =    2422.43 MB (highwater)       1453.69 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010104       0 memory =    2422.43 MB (highwater)       1453.73 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010105       0 memory =    2422.43 MB (highwater)       1453.77 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010106       0 memory =    2422.43 MB (highwater)       1455.78 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
(seq_mct_drv): ===============  pes min memory highwater  (MB)    2085.421  ===============
(seq_mct_drv): ===============  pes max memory highwater  (MB)    2422.431  ===============

 memory_write: model date =   00010102       0 memory =    2422.16 MB (highwater)       1453.43 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010103       0 memory =    2422.16 MB (highwater)       1453.49 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010104       0 memory =    2422.16 MB (highwater)       1453.51 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010105       0 memory =    2422.16 MB (highwater)       1453.54 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010106       0 memory =    2422.16 MB (highwater)       1455.67 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
(seq_mct_drv): ===============  pes min memory highwater  (MB)    2084.936  ===============
(seq_mct_drv): ===============  pes max memory highwater  (MB)    2422.163  ===============

 memory_write: model date =   00010102       0 memory =    2424.99 MB (highwater)       1455.07 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010103       0 memory =    2424.99 MB (highwater)       1455.09 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010104       0 memory =    2424.99 MB (highwater)       1455.16 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010105       0 memory =    2424.99 MB (highwater)       1455.20 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
 memory_write: model date =   00010106       0 memory =    2424.99 MB (highwater)       1457.46 MB (usage)  (pe=    0 comps= cpl ATM ICE GLC WAV IAC ESP)
(seq_mct_drv): ===============  pes min memory highwater  (MB)    2085.393  ===============
(seq_mct_drv): ===============  pes max memory highwater  (MB)    2424.991  ===============

```

**Comment by oksanaguba** [comment]
I am not familiar with this code. Would you please summarize what was wrong and what is the fix? thanks!

File: cime/src/share/util/shr_mem_mod.F90 [hunk 1]
```diff
@@ -91,8 +91,8 @@ subroutine shr_mem_getusage(r_msize,r_mrss,prt)
     !---------------------------------------------------
 
     ierr = GPTLget_memusage (msize, mrss, mshare, mtext, mdatastack)
-    r_msize = msize*mb_blk
-    r_mrss  = mrss*mb_blk
+    r_msize = msize / 1024_shr_kind_r8
+    r_mrss  = mrss  / 1024_shr_kind_r8
```
Comment (amametjanov) [inline cime/src/share/util/shr_mem_mod.F90]: @mt5555 just a note for your review, fortran now relies on mem-usage from C sys-calls.

**Comment by amametjanov** [comment]
> Would you please summarize what was wrong and what is the fix?

Previously, memory usage was estimated by allocating 8MB 64-bit real array and measuring memusage before and after. This estimate was way off on Anvil (must be CentOS7, other machines have OK estimates). Increasing the allocation on CPL master proc to 1GB improved the estimate, but it was still slightly off compared to sys-call results. New commits make sure that numbers in cpl.log (fortran) match with numbers in e3sm.log (C).

**Comment by AaronDonahue** [comment]
@oksanaguba , I see you still have a pending review.  Could you take another look and let me know if this PR is ready to merge?

**Comment by oksanaguba** [comment]
@AaronDonahue -- my understanding is that 1 review per PR is enough. But yes, I will click approve, thanks.

**Comment by AaronDonahue** [comment]
Oh, I actually didn't see that this had already been merged to next so it was on its way.  I'm going to merge to master now.

**Comment by oksanaguba** [comment]
Is it right to have 2 different integrators?

**Comment by AaronDonahue** [comment]
@rljacob ?

**Comment by rljacob** [comment]
This has one integrator (assignee) and 2 reviewers.  More then 1 reviewer is fine.

**Pull request merged**